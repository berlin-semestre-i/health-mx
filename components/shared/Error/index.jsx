import React from 'react'

export default ({ error }) => {
  const errorMessage = typeof error === 'string' ? error : 'An error occurred'

  return <div name="errorMessage">{errorMessage}</div>
}