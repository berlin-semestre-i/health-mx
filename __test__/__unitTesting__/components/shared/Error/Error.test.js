import React from 'react'
import { shallow, mount } from 'enzyme'
import Error from '../../../../../components/shared/Error'

const errorMessageString = 'error'
const errorMessageNotString = 2

describe('Error', () => {
  it('renders', () => {
    const component = shallow(<Error error={errorMessageString}/>)
    expect(component).toMatchSnapshot()
  })

  it('renders not string error correctly', () => {
    const component = mount(<Error error={errorMessageNotString}/>)
    expect(component.find('div[name="errorMessage"]').text()).toBe('An error occurred')
  })
})