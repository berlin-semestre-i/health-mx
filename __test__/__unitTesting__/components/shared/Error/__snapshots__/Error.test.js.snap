// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error renders 1`] = `
<div
  name="errorMessage"
>
  error
</div>
`;
